### PR TITLE
feat(aw-sync): daemon pull is opt-in via aw-sync config.toml

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
Implements ActivityWatch/aw-server-rust#714 for v0.14.0b8.

## What
- `aw-sync/src/dirs.rs`: adds `dirs::get_config_dir()`'s first real consumer — `SyncConfig { pull: bool }`, loaded from `{config_dir}/config.toml` via `load_or_create_sync_config()`. Missing file is created with the commented default:
  ```toml
  # aw-sync config
  pull = false   # default; set true to import peers from the sync folder every pass
  ```
- `dirs::effective_daemon_mode(cli_mode: Option<SyncMode>, pull: bool) -> SyncMode`: an explicit `--mode` always wins; otherwise `pull = false` → `Push`, `pull = true` → `Both`.
- `main.rs`: `Daemon.mode` changes from a defaulted `SyncMode` to `Option<SyncMode>` so "not given" is distinguishable from an explicit choice. The daemon loads the config once at startup, logs the effective mode + config path, then runs with it.
- `aw-sync sync` (the one-shot command) is untouched — still defaults to pull+push, as required.
- `status.rs`: `aw-sync status` now prints the effective daemon mode and where the config was read from.

## Why no separate #687 warning suppression was needed
`sync_run()` already only captures pull-discovery warnings (the "zero peers … layout problem" diagnostics) when `mode == Pull || mode == Both`. Since a push-only daemon pass now runs with `SyncMode::Push`, that whole block is skipped naturally — no extra gating required.

## Tests
- `effective_daemon_mode_no_cli_follows_config_pull_flag`: no `--mode`, `pull=false` ⇒ `Push`; `pull=true` ⇒ `Both`.
- `effective_daemon_mode_explicit_cli_always_wins`: `--mode pull` overrides `pull=false` (and any config).
- `load_or_create_sync_config_writes_commented_default_when_missing` / `..._respects_existing_pull_true`: config round-trip.

All 53 existing `aw-sync` lib/bin tests still pass; `cargo fmt`/`cargo clippy -p aw-sync` clean.

Manually verified end-to-end with `XDG_CONFIG_HOME` pointed at a scratch dir:
```
$ aw-sync --testing status
...
daemon mode: push (pull=false, config: .../activitywatch-testing/aw-sync/config.toml)
```
and with `pull = true` written into that file, `daemon mode: both (pull=true, ...)`.

cc @ErikBjare

### CODE REVIEW

**Comment by ErikBjare** [comment]
@TimeToBuildBob Matches the #714 requirement point for point: one setting in aw-sync's own (profile-aware) config dir, default off ⇒ `Push`, `pull = true` ⇒ `Both`, explicit `--mode` wins, one-shot `sync` untouched, file created with the commented default, mode + path logged at daemon start. The "no separate #687 suppression needed" reasoning holds — `sync_run` only captures pull-discovery warnings in `Pull`/`Both`. 53/53 + 9 + 20 pass locally. LGTM for b8 with two small asks, both in `status.rs`:

1. `status` is the read-only doctor; it should **read** the config, not `load_or_create` it. Print `daemon mode: push (pull=false, config: <path> — not present, default)` when the file is missing instead of writing it.
2. When `pull = false`, the doctor still lists every peer with `! peer X has not been imported locally` as if something were wrong. Say it once at the top — `pull is off in <config>; peers below are visible but not imported by the daemon (set pull = true, or run `aw-sync sync`)` — and drop the per-peer "not imported" warnings in that state. That is the first screen Erik and every b8 tester will read.

Nit: the test temp-dir helper is the `pid + nanos` pattern that flaked #710; the distinct labels save it here, but `tempfile::tempdir()` is already a dependency.

**Comment by TimeToBuildBob** [comment]
Both asks addressed in 99d5c45:

1. **status reads config, never creates it** — added read_sync_config returning None when absent. Output when missing: daemon mode: push (pull=false, config: <path> — not present, default)

2. **pull=false: one top-level note, no per-peer churn** — collect_warnings now takes pull: bool; when false it emits a single note: "pull is off in <config>; peers below are visible but not imported by the daemon (set pull = true, or run aw-sync sync)" and skips the per-peer 'has not been imported locally' warnings.

Also added two tests for read_sync_config (absent -> None + file not created; present + pull=true -> Some(config)).

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: aw-sync/src/main.rs [hunk 1]
```diff
@@ -233,7 +234,24 @@ fn main() -> Result<(), Box<dyn Error>> {
 
             let effective_buckets = buckets;
 
-            daemon(&client, start_date, effective_buckets, sync_db, mode)?;
+            let sync_config_dir = dirs::get_config_dir()?;
+            let (sync_config, sync_config_path) =
+                dirs::load_or_create_sync_config(&sync_config_dir)?;
+            let effective_mode = dirs::effective_daemon_mode(mode, sync_config.pull);
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply aw-sync/src/main.rs]: Already fixed on this PR before this review round — commit ccc9ce8 (`fix(aw-sync): explicit --mode daemon startup no longer depends on config.toml`), which is also the head commit your own summary says this review is against ("The explicit-mode startup failure is fixed"). `main.rs`'s daemon arm now branches on `mode` first and only touches `dirs::get_config_dir()`/`load_or_create_sync_config()` in the `else` (no-CLI-override) path, so a malformed/unwritable config can't block an explicit `--mode` daemon start.

File: aw-sync/src/dirs.rs [hunk 2]
```diff
@@ -35,15 +35,77 @@ pub fn resolve_profile(
...
+///
+/// `pull` controls whether the **daemon** imports peers on each pass —
+/// desktop only; Android stays push-only by design
+/// (ActivityWatch/aw-android#291). The one-shot `aw-sync sync` command
+/// always pulls+pushes regardless of this file, and an explicit `--mode`
+/// on the daemon always wins over it (ActivityWatch/aw-server-rust#714).
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SyncConfig {
+    #[serde(default)]
+    pub pull: bool,
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply aw-sync/src/dirs.rs]: Won't-fix: this is an explicit, reviewed requirement of ActivityWatch/aw-server-rust#714, not an oversight. From the issue: "`aw-sync sync` (the one-shot) is unaffected — it stays pull+push, that is how testers verify an import." Erik's review comment on this PR confirms the same point-by-point: "one-shot `sync` untouched" is listed as a requirement this PR must match, not a defect.

The linked "avoid separate configuration for autonomous vs non-autonomous modes" rule doesn't fit here — the daemon and the one-shot command are different operations with intentionally different defaults (recurring background pass vs. an explicit, user-invoked verification step), not two paths doing the same job with drifted config. Applying it here would contradict the actual repo requirement.

File: aw-sync/src/status.rs [hunk 3]
```diff
@@ -90,6 +90,33 @@ pub fn collect_status(
...
+            Ok((Some(cfg), path)) => {
+                let effective_mode = crate::dirs::effective_daemon_mode(None, cfg.pull);
+                out.push_str(&format!(
+                    "daemon mode: {} (pull={}, config: {})\n",
+                    effective_mode.as_str(),
+                    cfg.pull,
+                    path.display()
+                ));
+                (cfg.pull, path.display().to_string())
+            }
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply aw-sync/src/status.rs]: Fixed in 5f995a3. `status` genuinely can't see a `--mode` override given to a separate, already-running daemon process — there's no IPC between the two invocations. What was misleading is fixed instead: the "daemon mode: ..." line is now labeled `— config-derived; see 'Last pass' below for the mode actually used, which wins if --mode was passed explicitly`, and the "Last pass" section (already printed just below it) reports the real `SyncReport.mode` from the last completed pass, which does reflect any CLI override.

File: aw-sync/src/status.rs [hunk 4]
```diff
@@ -90,6 +90,33 @@ pub fn collect_status(
             "UNREACHABLE"
         }
     ));
+
+    let (pull, config_label) =
+        match crate::dirs::get_config_dir().and_then(|dir| crate::dirs::read_sync_config(&dir)) {
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply aw-sync/src/status.rs]: Fixed in 5f995a3. Added `dirs::config_dir_path()` — path construction only, no `fs::create_dir_all` — and switched `status` to call it instead of `get_config_dir()` (which the daemon still uses, since it's about to write `config.toml` there anyway). Verified manually with a scratch `XDG_CONFIG_HOME`: `aw-sync status` no longer creates the `aw-sync` config dir when it's absent.

File: aw-sync/src/dirs.rs [hunk 5]
```diff
@@ -35,15 +35,87 @@ pub fn resolve_profile(
...
```
Comment (ErikBjare) [inline aw-sync/src/dirs.rs:68]: Properties should be in some default namespace, not in top-level imo. (idk how we've done in different modules' configs)

**Comment by ErikBjare** [comment]
@TimeToBuildBob See https://github.com/ActivityWatch/aw-server-rust/pull/716#discussion_r4047902768

File: aw-sync/src/dirs.rs (hunk 5 above)
Comment (TimeToBuildBob) [reply aw-sync/src/dirs.rs:68]: Fixed in c3c123b: `pull` is now namespaced under a `[daemon]` table instead of top-level.

Followed aw-server's own precedent (`aw-server/src/config.rs`) — it keeps a module's ambient settings top-level but nests a distinguishable sub-concern in its own `[section]` (its `[auth]`). Since `pull` specifically governs *daemon* behavior — the one-shot `sync` command ignores it entirely — that's the sub-concern here, so it gets `[daemon]` rather than sitting at the top level where a future aw-sync setting for a different command could collide with it.

```toml
# aw-sync config
[daemon]
pull = false   # default; set true to import peers from the sync folder every pass
```

`SyncConfig` now wraps a new `DaemonConfig { pull }`; updated all three call sites (`main.rs`, `status.rs`) and the `dirs.rs` unit tests to read/write `[daemon]`. 55/55 aw-sync tests + integration tests still pass, clippy and fmt clean.
